File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-fx-5-4/PremiumQuote.cpp
// PremiumQuote.cpp 
#include "PremiumQuote.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PremiumQuote::PremiumQuote(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //valueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* valueNode = xmlNode->FirstChildElement("value");

   if(valueNode){valueIsNull_ = false;}
   else{valueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- valueNode , address : " << valueNode << std::endl;
   #endif
   if(valueNode)
   {
      if(valueNode->Attribute("href") || valueNode->Attribute("id"))
      {
          if(valueNode->Attribute("id"))
          {
             valueIDRef_ = valueNode->Attribute("id");
             value_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(valueNode));
             value_->setID(valueIDRef_);
             IDManager::instance().setID(valueIDRef_,value_);
          }
          else if(valueNode->Attribute("href")) { valueIDRef_ = valueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { value_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(valueNode));}
   }

   //quoteBasisNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* quoteBasisNode = xmlNode->FirstChildElement("quoteBasis");

   if(quoteBasisNode){quoteBasisIsNull_ = false;}
   else{quoteBasisIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- quoteBasisNode , address : " << quoteBasisNode << std::endl;
   #endif
   if(quoteBasisNode)
   {
      if(quoteBasisNode->Attribute("href") || quoteBasisNode->Attribute("id"))
      {
          if(quoteBasisNode->Attribute("id"))
          {
             quoteBasisIDRef_ = quoteBasisNode->Attribute("id");
             quoteBasis_ = boost::shared_ptr<PremiumQuoteBasisEnum>(new PremiumQuoteBasisEnum(quoteBasisNode));
             quoteBasis_->setID(quoteBasisIDRef_);
             IDManager::instance().setID(quoteBasisIDRef_,quoteBasis_);
          }
          else if(quoteBasisNode->Attribute("href")) { quoteBasisIDRef_ = quoteBasisNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { quoteBasis_ = boost::shared_ptr<PremiumQuoteBasisEnum>(new PremiumQuoteBasisEnum(quoteBasisNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDecimal> PremiumQuote::getValue()
{
   if(!this->valueIsNull_){
        if(valueIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(valueIDRef_));
        }else{
             return this->value_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
boost::shared_ptr<PremiumQuoteBasisEnum> PremiumQuote::getQuoteBasis()
{
   if(!this->quoteBasisIsNull_){
        if(quoteBasisIDRef_ != ""){
             return boost::shared_static_cast<PremiumQuoteBasisEnum>(IDManager::instance().getID(quoteBasisIDRef_));
        }else{
             return this->quoteBasis_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PremiumQuoteBasisEnum>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-fx-5-4/PremiumQuote.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PremiumQuote : ISerialized
    {
        public PremiumQuote(XmlNode xmlNode)
        {
            XmlNode valueNode = xmlNode.SelectSingleNode("value");
            
            if (valueNode != null)
            {
                if (valueNode.Attributes["href"] != null || valueNode.Attributes["id"] != null) 
                {
                    if (valueNode.Attributes["id"] != null) 
                    {
                        valueIDRef_ = valueNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(valueNode);
                        IDManager.SetID(valueIDRef_, ob);
                    }
                    else if (valueNode.Attributes["href"] != null)
                    {
                        valueIDRef_ = valueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        value_ = new XsdTypeDecimal(valueNode);
                    }
                }
                else
                {
                    value_ = new XsdTypeDecimal(valueNode);
                }
            }
            
        
            XmlNode quoteBasisNode = xmlNode.SelectSingleNode("quoteBasis");
            
            if (quoteBasisNode != null)
            {
                if (quoteBasisNode.Attributes["href"] != null || quoteBasisNode.Attributes["id"] != null) 
                {
                    if (quoteBasisNode.Attributes["id"] != null) 
                    {
                        quoteBasisIDRef_ = quoteBasisNode.Attributes["id"].Value;
                        PremiumQuoteBasisEnum ob = new PremiumQuoteBasisEnum(quoteBasisNode);
                        IDManager.SetID(quoteBasisIDRef_, ob);
                    }
                    else if (quoteBasisNode.Attributes["href"] != null)
                    {
                        quoteBasisIDRef_ = quoteBasisNode.Attributes["href"].Value;
                    }
                    else
                    {
                        quoteBasis_ = new PremiumQuoteBasisEnum(quoteBasisNode);
                    }
                }
                else
                {
                    quoteBasis_ = new PremiumQuoteBasisEnum(quoteBasisNode);
                }
            }
            
        
        }
        
    
        #region Value_
        private XsdTypeDecimal value_;
        public XsdTypeDecimal Value_
        {
            get
            {
                if (this.value_ != null)
                {
                    return this.value_; 
                }
                else if (this.valueIDRef_ != null)
                {
                    value_ = IDManager.getID(valueIDRef_) as XsdTypeDecimal;
                    return this.value_; 
                }
                else
                {
                      return this.value_; 
                }
            }
            set
            {
                if (this.value_ != value)
                {
                    this.value_ = value;
                }
            }
        }
        #endregion
        
        public string valueIDRef_ { get; set; }
        #region QuoteBasis_
        private PremiumQuoteBasisEnum quoteBasis_;
        public PremiumQuoteBasisEnum QuoteBasis_
        {
            get
            {
                if (this.quoteBasis_ != null)
                {
                    return this.quoteBasis_; 
                }
                else if (this.quoteBasisIDRef_ != null)
                {
                    quoteBasis_ = IDManager.getID(quoteBasisIDRef_) as PremiumQuoteBasisEnum;
                    return this.quoteBasis_; 
                }
                else
                {
                      return this.quoteBasis_; 
                }
            }
            set
            {
                if (this.quoteBasis_ != value)
                {
                    this.quoteBasis_ = value;
                }
            }
        }
        #endregion
        
        public string quoteBasisIDRef_ { get; set; }
        
    
        
    
    }
    
}

